src/libstore/daemon.cc

```diff
@@ -276,8 +276,13 @@ static void performOp(TunnelLogger * logger, ref<Store> store,
 
     case wopQueryValidPaths: {
         auto paths = worker_proto::read(*store, from, Phantom<StorePathSet> {});
+        SubstituteFlag substitute = NoSubstitute;
+        if (GET_PROTOCOL_MINOR(clientVersion) >= 27) {
+            substitute = readInt(from) ? Substitute : NoSubstitute;
+        }
         logger->startWork();
-        auto res = store->queryValidPaths(paths);
+
```

Comment:
Not to be a nit-pick but I would rather not put a new blank line here because the `startWork`, `queryValidPaths`, and `stopWork` very much go together. Maybe if you want to add blank lines they could go before and after that group?

Reply:
Fixed :)